New Module: Listbox

ianmcburnie opened the pull request:
## Description

Added new Listbox module based on the [W3C Authoring Pattern](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html) (which we will also add to MIND patterns).

Added basic expand/collapse behaviour only to the website (I want to clean up & simplify the JS across DS4 and DS6 sites in a future PR).

## Context

Our readonly combobox has issues in iOS. This listbox module offers another alternative for creating a custom HTML select element.

The listbox borrows most of it's styling from existing "dropdown" styles, therefore this PR should be fairly straightforward. The listbox is styled exactly like the Menu module (makes sense, given that they both use an `expand-btn`). If we decide to make it more like the current Combobox, in terms of colours, then let's do that as a follow up PR as it may require changes to the button module (which I'd rather not do until the button PR is merged).

## References

Closes #396 

## Screenshots

Notice in the last two screenshots that there appears to be some issue with the Combobox overlapping the Listbox. I will look into this as part of #399.

![screen shot 2018-10-08 at 12 34 07 pm](https://user-images.githubusercontent.com/38065/46632132-7b10c280-cafe-11e8-8f4a-44960fd24b06.png)
![screen shot 2018-10-08 at 12 34 17 pm](https://user-images.githubusercontent.com/38065/46632133-7b10c280-cafe-11e8-9201-4f08c910bd1a.png)
![screen shot 2018-10-08 at 12 34 21 pm](https://user-images.githubusercontent.com/38065/46632135-7ba95900-cafe-11e8-83e7-a1d6a6e6510c.png)
![screen shot 2018-10-08 at 12 34 33 pm](https://user-images.githubusercontent.com/38065/46632137-7ba95900-cafe-11e8-9fa7-bbd7e0b2bb77.png)
![screen shot 2018-10-08 at 12 34 38 pm](https://user-images.githubusercontent.com/38065/46632138-7ba95900-cafe-11e8-87bd-2a15b76edd62.png)
![screen shot 2018-10-08 at 12 34 43 pm](https://user-images.githubusercontent.com/38065/46632140-7ba95900-cafe-11e8-910c-f8519c457ad2.png)
![screen shot 2018-10-08 at 12 35 06 pm](https://user-images.githubusercontent.com/38065/46632141-7ba95900-cafe-11e8-9c19-8ea7eff27b98.png)
![screen shot 2018-10-08 at 12 35 18 pm](https://user-images.githubusercontent.com/38065/46632142-7c41ef80-cafe-11e8-9f55-e35eb33b734b.png)

seangates commented on docs/test/misc/ds6/alignment-controls-inline.html:
```diff
@@ -75,13 +97,37 @@ <h2>Medium</h2>
         </div>
     </span>
+    <span class="listbox">
+        <button class="expand-btn expand-btn--medium" aria-expanded="false" aria-haspopup="listbox">
```
Looks like the `expand-btn--regular` needs some adjusting:

![image](https://user-images.githubusercontent.com/105656/46686567-7fe07f80-cbad-11e8-9839-ffe2c3139c0c.png)

seangates commented:
Otherwise, looks good!

ianmcburnie commented:
Added missing padding property for `expand-btn--medium`. I see some duplicate rules for padding in the less file, but want to keep my changes to a minimum so I don't cause conflicts with your button PR. I also added the menu to the test page.

![screen shot 2018-10-09 at 12 54 36 pm](https://user-images.githubusercontent.com/38065/46694983-1e2b1000-cbc3-11e8-8839-247944c062cb.png)

![screen shot 2018-10-09 at 12 54 28 pm](https://user-images.githubusercontent.com/38065/46694989-23885a80-cbc3-11e8-9b9b-388f84122ab4.png)

seangates commented:
@ianmcburnie Well ... these files all change with the #401 PR.
